src/Client/Campaign.php

```diff
@@ -17,6 +17,11 @@ public function getById(string $id): array
     {
         try {
             $response = $this->getHttpClient()->get("{$this->getSetting('campaign', 'baseUri')}/$id");
+
+            if (!json_decode($response->getBody()->getContents(), true)['ready']) {
```

Comment:
This part cannot lead to a `RequestException` so it would sit more naturally and leave the code more readable outside the `try/catch`.

Additionally, `json_decode()` of the whole response doesn't need to happen twice. We can assign the existing decode result to a local var and then check that.

I think it would also be prudent to check if the property is set so we don't need to worry about the order changes are released. If the property is absent we should assume we have an older version of the API contract and not exclude the campaign.